src/Components/Browser.JS/src/Platform/Circuits/CircuitHandler.ts

```diff
@@ -0,0 +1,7 @@
+export abstract class CircuitHandler {
```

Comment:
I think this would be better as an `interface`, not `abstract class`.

That's because we're not giving people a way of deriving from our class anyway (we don't even expose type information). Using an `interface` more closely matches what we're really doing here, which is looking for specific method names, not a specific base class.

Reply:
Also, if it's an interface, it would be good for its members to be optional. People might want to implement a subset of them.